crates/datasources/src/json/stream.rs

```diff
@@ -1,97 +1,88 @@
 use std::pin::Pin;
-use std::sync::{Arc, Mutex};
+use std::sync::Arc;
 use std::task::{Context, Poll};
 
 use datafusion::arrow::datatypes::{Schema, SchemaRef};
 use datafusion::arrow::json::ReaderBuilder;
 use datafusion::arrow::record_batch::RecordBatch;
-use datafusion::error::DataFusionError;
+use datafusion::error::{DataFusionError, Result as DFResult};
 use datafusion::execution::TaskContext;
 use datafusion::physical_plan::streaming::PartitionStream;
 use datafusion::physical_plan::{RecordBatchStream, SendableRecordBatchStream};
-use futures::{Stream, StreamExt};
+use futures::channel::mpsc;
+use futures::{SinkExt, Stream, StreamExt, TryStreamExt};
+use json_stream::JsonStream;
 use object_store::{ObjectMeta, ObjectStore};
 use serde_json::{Map, Value};
 
-use crate::json::errors::JsonError;
-use crate::json::table::push_unwind_json_values;
+use crate::json::errors::{JsonError, Result};
 
-pub type SendableCheckedRecordBatchStream =
-    Pin<Box<dyn Stream<Item = Result<RecordBatch, DataFusionError>> + Send>>;
-
-pub struct JsonStream {
+pub struct JsonRecordBatchStream {
     schema: Arc<Schema>,
-    stream: SendableCheckedRecordBatchStream,
+    // this is the same as a sendable recordbatch stream, but declared
+    // separtley so we can have isomorphic values using adapters.
+    stream: Pin<Box<dyn Stream<Item = DFResult<RecordBatch>> + Send>>,
```

Comment:
I don't know what this means.

Reply:
It might have been from not importing the `futures::stream::Stream` trait since those methods are defined on the trait.